### ISSUE TITLE
xdcc not work for more than 4GB

### ISSUE THREAD

**Issue by bibiyanki** [irssi/irssi#977]
hello:
i try download for more than 4gb with irssi and xdcc but the bot disconnect me

what's wrong???

**Comment by vague666** [comment]
What's the filesystem you are saving the file to and are you on a 32bit OS?

**Comment by bibiyanki** [comment]
i saving on home/user/ and my system are deepin 64bits (it's debian)

**Comment by vague666** [comment]
home/user might be a mount point but it doesn't tell us what the filesystem is

**Comment by bibiyanki** [comment]
it's zip file i try to save

**Comment by vague666** [comment]
zip is the file contents, what is the file<b>system</b>(fat, ntfs, ext3, ext4 etc) you are writing to

**Comment by wilhelmy** [comment]
IIRC this is a known problem with large DCC file transfers?

**Comment by ailin-nemui** [comment]
Closing for inactivity

**Comment by Poorchop** [comment]
I have been having this issue with files >4 GB. Although it does download most of the file, the download always fails at the last few bytes. It states the connection has been reset by the peer. So far, I have tried this on an APFS filesystem on a 64-bit OS. I can try this on btrfs at some point but I don't suspect that the filesystem is the issue.

I believe that these have been passive DCC transfers but I also don't think that's the issue. For what it's worth, irssi is not the only client that has this issue—I did try one other that also always failed at the last few bytes, but that was a different environment on an NTFS filesystem.
